File: Examples/ObjcDemo/RixSDKPodDemo/RixEngineObjcDemo/UI/Max/MaxMainVC.m
#import "MaxMainVC.h"
#import "MaxBannerVC.h"
#import "MaxRewardVideoVC.h"
#import "MaxInterstitialVC.h"
#import "MaxNativeVC.h"
#import <AppLovinSDK/AppLovinSDK.h>
#import "AdConfig.h"

@implementation MaxMainVC

- (void)viewDidLoad {
    [super viewDidLoad];
    self.navigationItem.title = NSLocalizedString(@"max_ad", @"");
}

- (MenuAppearance)menuAppearance {
    return MenuAppearanceCard;
}

- (NSArray<MenuItem *> *)menuItems {
    return @[
        [[MenuItem alloc] initWithTitle:NSLocalizedString(@"banner_ad", @"") makeVC:^UIViewController *{ return [[MaxBannerVC alloc] init]; }],
        [[MenuItem alloc] initWithTitle:NSLocalizedString(@"rewardVideo_ad", @"") makeVC:^UIViewController *{ return [[MaxRewardVideoVC alloc] init]; }],
        [[MenuItem alloc] initWithTitle:NSLocalizedString(@"interstitial_ad", @"") makeVC:^UIViewController *{ return [[MaxInterstitialVC alloc] init]; }],
        [[MenuItem alloc] initWithTitle:NSLocalizedString(@"native_ad", @"") makeVC:^UIViewController *{ return [[MaxNativeVC alloc] init]; }]
    ];
}

- (NSString *)menuSubtitleAtIndex:(NSInteger)index {
    NSArray *subtitles = @[
        @"Flexible formats at the top, middle or bottom of your app.",
        @"Users engage with a video ad in exchange for in-app rewards.",
        @"Full-screen ads at natural breaks or transition points.",
        @"Ads that match the look and feel of your app."
    ];
    if (index >= 0 && index < subtitles.count) {
        return subtitles[index];
    }
    return nil;
}

- (void)setupSDK {
    ALSdkInitializationConfiguration *initConfig = [ALSdkInitializationConfiguration configurationWithSdkKey:[AdConfig Max_App_Key] builderBlock:^(ALSdkInitializationConfigurationBuilder *builder) {
        builder.mediationProvider = ALMediationProviderMAX;
    }];

    ALSdkSettings *settings = [ALSdk shared].settings;
    [settings setExtraParameterForKey:@"uid2_token" value:@"liuweileliuweile"];

    // Note: ALPrivacySettings might be deprecated or moved in newer SDKs, 
    // but following the Swift version's logic.
    // [ALPrivacySettings setDoNotSell:NO];
    // [ALPrivacySettings setHasUserConsent:YES];

    [[ALSdk shared] initializeWithConfiguration:initConfig completionHandler:^(ALSdkConfiguration *config) {
        // Initialization complete
    }];
}

@end

File: AlxAdsDemo/UI/Max/MaxMainVC.swift
//
//  MaxMainVC.swift
//  AdsDemo
//
//  Created by liu weile on 2023/6/2.
//

import UIKit
import AppLovinSDK

class MaxMainVC: BaseMenuVC {

    override var menuItems: [MenuItem] {[
        (title: NSLocalizedString("banner_ad",       comment: ""), makeVC: { MaxBannerVC() }),
        (title: NSLocalizedString("rewardVideo_ad",  comment: ""), makeVC: { MaxRewardVideoVC() }),
        (title: NSLocalizedString("interstitial_ad", comment: ""), makeVC: { MaxInterstitialVC() }),
        (title: NSLocalizedString("native_ad",       comment: ""), makeVC: { MaxNativeVC() }),
    ]}

    override func viewDidLoad() {
        super.viewDidLoad()
        navigationItem.title = NSLocalizedString("max_ad", comment: "")
    }

    override var menuAppearance: MenuAppearance { .card }

    override func menuSubtitle(at index: Int) -> String? {
        let subtitles = [
            "Flexible formats at the top, middle or bottom of your app.",
            "Users engage with a video ad in exchange for in-app rewards.",
            "Full-screen ads at natural breaks or transition points.",
            "Ads that match the look and feel of your app."
        ]
        guard index >= 0, index < subtitles.count else { return nil }
        return subtitles[index]
    }

    override func setupSDK() {
        let initConfig = ALSdkInitializationConfiguration(sdkKey: AdConfig.Max_App_Key) { builder in
            builder.mediationProvider = ALMediationProviderMAX
        }

        let settings = ALSdk.shared().settings
        settings.setExtraParameterForKey("uid2_token", value: "liuweileliuweile")

        ALPrivacySettings.setDoNotSell(false)
        ALPrivacySettings.setHasUserConsent(true)

        ALSdk.shared().initialize(with: initConfig) { _ in }
    }
}
